internal/cmd/rsl/record/record.go

```diff
@@ -12,20 +14,32 @@ type options struct{}
 
 func (o *options) AddFlags(_ *cobra.Command) {}
 
-func (o *options) Run(_ *cobra.Command, args []string) error {
+func (o *options) Run(cmd *cobra.Command, args []string) error {
 	repo, err := repository.LoadRepository()
 	if err != nil {
 		return err
 	}
 
-	return repo.RecordRSLEntryForReference(args[0], true)
+	if err := repo.RecordRSLEntryForReference(args[0], true); err != nil {
+		return err
+	}
+
+	if err := repo.VerifyRef(cmd.Context(), args[0], false); err != nil {
+		if err := repo.InvalidateLatestRSLEntry(args[0], true); err != nil {
```

Comment:
Hm, I think we want to rethink this. The RSL reference entry represents the intent to _push_ and make the change to the branch (say main) available to other users. At this point, the entry's been created but no one other than this local client is aware of it. So if the entry is invalid, we should actually `gitinterface.Reset` the RSL branch and remove that entry altogether.

Reply:
Fixed